CHANGELOG.md

```diff
@@ -1,27 +1,28 @@
-# Release History
+# CHANGELOG 📝
 
-## 0.4.2 (Unreleased)
+The Azure MCP Server updates automatically by default whenever a new release comes out 🚀. We ship updates twice a week on Tuesdays and Thursdays 😊
+
+## 0.4.2 (2025-07-22)
 
 ### Features Added
 
+- Added support for installation as a VS Code extension as an alternative to using NPX or Docker (Microsoft Container Registry). [[#661](https://github.com/Azure/azure-mcp/pull/661)]
 - Added more tools for Azure Key Vault: [[#517](https://github.com/Azure/azure-mcp/pull/517)]
-  - `azmcp-keyvault-certificate-list`
-  - `azmcp-keyvault-certificate-get`
-  - `azmcp-keyvault-certificate-create`
-  - `azmcp-keyvault-secret-list`
-  - `azmcp-keyvault-secret-create`
-- Added support for Azure Workbooks management operations:
+  - `azmcp-keyvault-certificate-list` - List certificates in a Key Vault
+  - `azmcp-keyvault-certificate-get` - Get details of a specific certificate
+  - `azmcp-keyvault-certificate-create` - Create a new certificate in a Key Vault
+  - `azmcp-keyvault-secret-list` - List secrets in a Key Vault
+  - `azmcp-keyvault-secret-create` - Create a new secret in a Key Vault
+- Added support for Azure Workbooks management operations: [[#629](https://github.com/Azure/azure-mcp/pull/629)]
   - `azmcp-workbooks-list` - List workbooks in a resource group with optional filtering
   - `azmcp-workbooks-show` - Get detailed information about a specific workbook
   - `azmcp-workbooks-create` - Create new workbooks with custom visualizations and content
   - `azmcp-workbooks-update` - Update existing workbook configurations and metadata
   - `azmcp-workbooks-delete` - Delete workbooks when no longer needed
 
-### Breaking Changes
-
 ### Bugs Fixed
 
-- Fixes issue where tool call can fail if MCP host does not first list tools [[#556](https://github.com/Azure/azure-mcp/issues/556)]
+- Fixes issue where tool call can fail if MCP host does not first list tools. [[#556](https://github.com/Azure/azure-mcp/issues/556)]
```

Comment:
```suggestion
- Fixes tool discovery race condition causing "tool not found" errors in MCP clients like LangGraph [[#556](https://github.com/Azure/azure-mcp/issues/556)]
```

Reply:
A slightly detailed changelog entry would be: Fixes the "tool not found" error that occurs when the MCP client (like Langchain python library) uses a different MCP session to invoke the tool than it used to list available tools.